### PR TITLE
[MIG][12.0] account_invoice_supplierinfo_update_discount and triple_discount (from 8.0)

### PR DESCRIPTION
**Pull request by legalsylvain**
Migrate ``account_invoice_supplierinfo_update_discount`` from 8.0 to 12.0 It's a glue module that allow to update the ``discount`` of the ``supplierinfo`` from the invoice lines, when ``purchase_discount`` and ``account_invoice_supplierinfo_update`` are installed. 

In the same way add the trivial module ``account_invoice_supplierinfo_update_triple_discount`` for the similar purpose, that is a glue module for ``purchase_triple_discount``, ``account_invoice_supplierinfo_update`` and ``account_invoice_triple_discount``.

~Please review first main PR that port ``account_invoice_supplierinfo_update`` from 10 to 12 here, https://github.com/OCA/account-invoicing/pull/578.~ (Merged)

#GRAPOCA

### CODE REVIEW

File: account_invoice_supplierinfo_update_discount/wizard/wizard_update_invoice_supplierinfo_line.py
Comment (Trivedi-Vacha-SerpentCS) [inline account_invoice_supplierinfo_update_discount/wizard/wizard_update_invoice_supplierinfo_line.py]: Import statement should be in alphabetical order

File: account_invoice_supplierinfo_update_triple_discount/models/account_invoice_line.py
Comment (Trivedi-Vacha-SerpentCS) [inline account_invoice_supplierinfo_update_triple_discount/models/account_invoice_line.py]: from odoo import api, models

File: account_invoice_supplierinfo_update_triple_discount/wizard/wizard_update_invoice_supplierinfo_line.py
Comment (Trivedi-Vacha-SerpentCS) [inline account_invoice_supplierinfo_update_triple_discount/wizard/wizard_update_invoice_supplierinfo_line.py]: same as above

Comment (legalsylvain) [reply account_invoice_supplierinfo_update_triple_discount/wizard/wizard_update_invoice_supplierinfo_line.py]: done.

File: account_invoice_supplierinfo_update_triple_discount/models/account_invoice_line.py
Comment (legalsylvain) [reply account_invoice_supplierinfo_update_triple_discount/models/account_invoice_line.py]: done.

File: account_invoice_supplierinfo_update_discount/wizard/wizard_update_invoice_supplierinfo_line.py
Comment (legalsylvain) [reply account_invoice_supplierinfo_update_discount/wizard/wizard_update_invoice_supplierinfo_line.py]: Done.

**Comment by legalsylvain** [comment]
Hi @Trivedi-Vacha-SerpentCS. I took into account your remarks. Could you update your review ? 

thanks !

**Comment by legalsylvain** [comment]
Travis fails for unrelated reasons.
See : https://github.com/OCA/account-invoicing/issues/693

**Review by quentinDupont — APPROVED** [review]
Works just as excepted :heavy_check_mark:

**Comment by legalsylvain** [comment]
Hi @Trivedi-Vacha-SerpentCS. could you update your review ? 

thanks !

**Comment by mourad-ehm** [comment]
Why there is not tests for account_invoice_supplierinfo_update_discount ?

**Comment by legalsylvain** [comment]
Hi @mourad-ehm. Thanks a lot for your review. I added a test for the module ``account_invoice_supplierinfo_update_discount``. Could you update your review ? 

thanks !

**Review by mourad-ehm — APPROVED** [review]

File: account_invoice_supplierinfo_update_discount/tests/test_module.py
Comment (chafique-delli) [inline account_invoice_supplierinfo_update_discount/tests/test_module.py]: @legalsylvain , the type of journal is not 'purchase' for the purchase invoice?

File: account_invoice_supplierinfo_update_triple_discount/tests/test_module.py
Comment (chafique-delli) [inline account_invoice_supplierinfo_update_triple_discount/tests/test_module.py]: @legalsylvain , the type of journal is not 'purchase' for the purchase invoice?

File: account_invoice_supplierinfo_update_discount/tests/test_module.py
Comment (legalsylvain) [reply account_invoice_supplierinfo_update_discount/tests/test_module.py]: Hi @chafique-delli. It doesn't change nothing, but indeed, it is not clean. 

I fixed it.

thanks !

File: account_invoice_supplierinfo_update_triple_discount/tests/test_module.py
Comment (legalsylvain) [reply account_invoice_supplierinfo_update_triple_discount/tests/test_module.py]: also fixed.

**Comment by legalsylvain** [comment]
Hi @chafique-delli. I took into account your remarks. Could you update your reviews ? 

thanks !

**Review by chafique-delli — APPROVED** [review]

**Comment by legalsylvain** [comment]
Thanks ! 

/ocabot merge patch

**Comment by legalsylvain** [comment]
damn'it ! I loose my superpower of OCA administrator.

I propose me as maintainers in the last commit, for the 3 modules GRAP co-developped.

 @OCA/accounting-maintainers : could you merge this approved PR ? 

thanks !

**Comment by houssine78** [comment]
Hi @legalsylvain do you think we can hope to have this PR merged ? what's blocking ? thanks !

**Comment by legalsylvain** [comment]
@houssine78  : 
- make an approval on the PR.
- ping maintainers @OCA/accounting-maintainers

**Comment by houssine78** [comment]
@legalsylvain I can't approved but I confirme it has been tested and approved from our side.

**Comment by legalsylvain** [comment]
you can approve. Click on "File changed", then click on "approve".

**Comment by houssine78** [comment]
Hi @jgrandguillaume @sbidoul do you think this can be merged ? it got the approval and it's just waiting to be merged.

Thank you

**Review by houssine78 — APPROVED** [review]

**Comment by sergiocorato** [comment]
@legalsylvain Usually there is 1 PR for module, at least 1 commit for module if not splittable, could you do it? (v.8 translation will be overridden I think, I don't experienced it)

**Comment by legalsylvain** [comment]
Hi @sergiocorato. Thanks for your review. I did the changes. 
- one commit for account_invoice_supplierinfo_update_discount migration
- one commit for account_invoice_supplierinfo_update_triple_discount

Indeed I usually make 2 PR, but the two modules comes together, has similar features to review, and have dependencies.

regarding the translation, I don't know. I just used the standard procedure. https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-12.0#technical-method-to-migrate-a-module-from-110-to-120-branch

@OCA/accounting-maintainers : I think this PR could be merged.

kind regards.

**Review by sergiocorato — APPROVED** [review]

**Comment by legalsylvain** [comment]
thanks to all for your reviews !

**Pull request merged**
